R/server.R

```diff
@@ -69,8 +70,41 @@ addResourcePath <- function(prefix, directoryPath) {
     getShinyOption("server")$setStaticPath(.list = stats::setNames(normalizedPath, prefix))
   }
 
-  # .globals$resourcePaths persists across runs of applications.
+  # .globals$resourcePaths and .globals$resources persist across runs of applications.
   .globals$resourcePaths[[prefix]] <- staticPath(normalizedPath)
+  # This is necessary because resourcePaths is only for serving assets out of C++;
+  # to support subapps, we also need assets to be served out of R, because those
+  # URLs are rewritten by R code (i.e. routeHandler) before they can be matched to
+  # a resource path.
+  .globals$resources[[prefix]] <- list(
+    directoryPath = normalizedPath,
+    func = staticHandler(normalizedPath)
+  )
+}
+
+resourcePathHandler <- function(req) {
+  if (!identical(req$REQUEST_METHOD, 'GET'))
```

Comment:
Just checking: when the original value of `req$REQUEST_METHOD` is `HEAD`, it would have been converted to GET by this point, right?

Reply:
That's right, it's way up at the outer edge of the `call` method.